compiler/rustc_lint/src/builtin.rs

```diff
@@ -1232,10 +1232,26 @@ impl<'tcx> LateLintPass<'tcx> for MutableTransmutes {
 }
 
 declare_lint! {
-    /// The `unstable_features` is deprecated and should no longer be used.
+    /// The `unstable_features` lint detects uses of `#![feature]`.
+    ///
+    /// ### Example
+    ///
+    /// ```rust,compile_fail
+    /// #![deny(unstable_features)]
+    /// #![feature(test)]
+    /// ```
+    ///
+    /// {{produces}}
+    ///
+    /// ### Explanation
+    ///
+    /// Crates that use unstable features require a nightly compiler which might not be desirable.
+    /// Especially in larger projects with a multitude of crates where some crates are allowed to
+    /// use unstable features and some are not, this lint may come in handy to enforce policies of
+    /// this kind.
```

Comment:
maybe add an explicit note that this is only necessarily when the project is normally built using a nightly compiler and irrelevant for anyone using the stable compiler?